ksqldb-engine/src/main/java/io/confluent/ksql/engine/EngineExecutor.java

```diff
@@ -511,12 +506,36 @@ private boolean isSourceTableMaterializationEnabled() {
         .getBoolean(KsqlConfig.KSQL_SOURCE_TABLE_MATERIALIZATION_ENABLED);
   }
 
+  private Optional<SourceName> getSourceNameIfIsNotExists(final ConfiguredStatement<?> statement) {
+    if (statement.getStatement() instanceof CreateStream
+        && ((CreateStream) statement.getStatement()).isNotExists()) {
+      return Optional.ofNullable(((CreateStream) statement.getStatement()).getName());
+    } else if (statement.getStatement() instanceof CreateTable
+        && ((CreateTable) statement.getStatement()).isNotExists()) {
+      return Optional.ofNullable(((CreateTable) statement.getStatement()).getName());
+    } else if (statement.getStatement() instanceof CreateTableAsSelect
+        && ((CreateTableAsSelect) statement.getStatement()).isNotExists()) {
+      return Optional.ofNullable(((CreateAsSelect) statement.getStatement()).getName());
+    } else if (statement.getStatement() instanceof CreateStreamAsSelect
+        && ((CreateStreamAsSelect) statement.getStatement()).isNotExists()) {
+      return Optional.ofNullable(((CreateAsSelect) statement.getStatement()).getName());
+    }
```

Comment:
We can reduce this by having the `isNotExists` in the `CreateSource` and `CreateAsSelect` classes instead.